x-pack/plugins/fleet/public/applications/fleet/sections/debug/components/agent_policy_debugger.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiCode,
+  EuiComboBox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+import { useQuery } from 'react-query';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { sendGetAgentPolicies, useLink } from '../../../hooks';
+import { SO_SEARCH_LIMIT } from '../../../constants';
+
+import { policyHasFleetServer } from '../../../services';
+import type { AgentPolicy } from '../../../types';
+import { AgentPolicyDeleteProvider } from '../../agent_policy/components';
+
+import { queryClient } from '..';
+
+import { CodeBlock } from './code_block';
+
+const fetchAgentPolicies = async () => {
+  const response = await sendGetAgentPolicies({
+    full: true,
+    perPage: SO_SEARCH_LIMIT,
+    sortOrder: 'asc',
+  });
+
+  if (response.error) {
+    throw new Error(response.error.message);
+  }
+
+  return response;
+};
+
+export const AgentPolicyDebugger: React.FunctionComponent = () => {
+  const { getHref } = useLink();
+  const [selectedPolicyId, setSelectedPolicyId] = useState<string>();
+
+  // TODO: Depending on the number of agent policies, this might need to be switched to
+  // `useInfinite` query with an infinite scrolling approach in the dropdown options.
+  const { data, status } = useQuery('debug-agent-policies', fetchAgentPolicies);
+
+  const agentPolicies = data?.data?.items ?? [];
+  const comboBoxOptions = agentPolicies.map((policy) => ({
+    label: `${policy.name} - ${policy.id}`,
+    value: policy.id,
+  }));
+
+  const selectedOptions = selectedPolicyId
+    ? [comboBoxOptions.find((option) => option.value === selectedPolicyId)!]
+    : [];
+
+  const selectedAgentPolicy = agentPolicies.find((policy) => policy.id === selectedPolicyId);
+
+  const onDelete = () => {
+    setSelectedPolicyId(undefined);
+    queryClient.invalidateQueries('debug-agent-policies');
+  };
+
+  if (status === 'error') {
+    return (
+      <EuiCallOut title="Error" color="danger">
+        Error fetching Agent Policies
+      </EuiCallOut>
+    );
+  }
+
+  return (
+    <>
+      <EuiText grow={false}>
+        <p>
+          Search for an Agent Policy using its name or <EuiCode>id</EuiCode> value. Use the code
+          block below to diagnose any potential issues with the policy{"'"}s configuration.
```

Comment:
Shouldn't this be translated? I'm not repeating the comment everywhere but it should be done to all the text in the page, unless we decide to keep it in english.

Reply:
Hmm you are probably correct that these should have translations. Good call.